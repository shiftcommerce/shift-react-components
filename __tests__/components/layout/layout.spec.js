// Libraries
import React from 'react'

// Components
import CustomHead from '../../../src/components/layout/custom-head'
import Layout from '../../../src/components/layout/layout'
import Minibag from '../../../src/components/layout/minibag'
import NavBar from '../../../src/components/navigation/navbar'

// Objects
import Logo from '../../../src/objects/logo'

// Fixtures
import menu from '../../fixtures/menu'

test('renders the header', () => {
  // Arrange
  const initialState = {
    shrunk: false,
    toggleShowClass: false,
    skipHeader: true,
    menu: menu
  }

  // Act
  const wrapper = shallow(
    <Layout
      router={{ pathname: '/' }}
      shrunk={initialState.shrunk}
      ShowClass={initialState.toggleShowClass}
      skipHeader={initialState.skipHeader}
      {...initialState}
    />
  )

  const header = wrapper.find('div.o-header')

  // Assert
  expect(header).toExist()
  expect(header).toContainReact(<Logo className='o-header__logo' />)
  expect(header.find(Minibag)).toExist()
  expect(wrapper).toMatchSnapshot()
})

test('renders the navbar for non-checkout pages', () => {
  // Arrange
  const initialState = {
    shrunk: false,
    toggleShowClass: false,
    skipHeader: true,
    menu: menu
  }

  // Act
  const wrapper = shallow(
    <Layout
      router={{ pathname: '/' }}
      shrunk={initialState.shrunk}
      ShowClass={initialState.toggleShowClass}
      skipHeader={initialState.skipHeader}
      {...initialState}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find(NavBar)).toExist()
})

test('renders the footer', () => {
  // Act
  const wrapper = shallow(
    <Layout router={{ pathname: '/' }} />
  )

  // Assert
  expect(wrapper.find('div.o-footer')).toIncludeText('Footer')
  expect(wrapper).toMatchSnapshot()
})

test('renders the children inside it', () => {
  // Arrange
  const initialState = {
    shrunk: false,
    toggleShowClass: false,
    skipHeader: true,
    menu: menu
  }

  // Act
  const wrapper = shallow(
    <Layout
      router={{ pathname: '/' }}
      shrunk={initialState.shrunk}
      ShowClass={initialState.toggleShowClass}
      skipHeader={initialState.skipHeader}
      {...initialState} >
      Content
    </Layout >
  )

  // Assert
  expect(wrapper.find('div.o-body')).toIncludeText('Content')
  expect(wrapper).toMatchSnapshot()
})

test('for checkout pages renders the custom head only', () => {
  // Arrange
  const initialState = {
    shrunk: false,
    toggleShowClass: false,
    skipHeader: false,
    menu: menu
  }

  // Act
  const wrapper = shallow(
    <Layout
      router={{ pathname: '/checkout/shipping-address' }}
      shrunk={initialState.shrunk}
      ShowClass={initialState.toggleShowClass}
      skipHeader={initialState.skipHeader}
      {...initialState}
    />
  )

  // Assert
  expect(wrapper.find(NavBar).exists()).toBe(false)
  expect(wrapper.find(CustomHead).exists()).toBe(false)
})
