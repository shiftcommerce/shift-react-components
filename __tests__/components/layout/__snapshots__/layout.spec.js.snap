// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the children inside it 1`] = `
<Fragment>
  <div
    className="o-header"
  >
    <div
      className="o-header__top"
    >
      <div
        className="o-header__top-wrapper"
      >
        <Logo
          className="o-header__logo"
        />
        <span
          className="o-nav u-hidden-d"
        >
          <NavBar
            menu={
              Object {
                "error": false,
                "id": "14",
                "loading": false,
                "menu_items": Array [
                  Object {
                    "canonical_path": "/trays",
                    "id": "54",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Trays",
                  },
                  Object {
                    "canonical_path": "/nonslip-trays",
                    "id": "55",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Nonslip trays",
                  },
                  Object {
                    "canonical_path": "/jugs",
                    "id": "56",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Jugs",
                  },
                  Object {
                    "canonical_path": "/filters",
                    "id": "57",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Filters",
                  },
                  Object {
                    "canonical_path": "/tea",
                    "id": "58",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Tea",
                  },
                  Object {
                    "canonical_path": "/saucers",
                    "id": "60",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Saucers",
                  },
                  Object {
                    "canonical_path": "/cups",
                    "id": "59",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Cups",
                  },
                  Object {
                    "canonical_path": "/coffee",
                    "id": "61",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Coffee",
                  },
                  Object {
                    "canonical_path": "/servers",
                    "id": "63",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Servers",
                  },
                  Object {
                    "canonical_path": "/mugs",
                    "id": "62",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Mugs",
                  },
                ],
                "pagination": Object {
                  "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
                  "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
                  "page_count": 1,
                  "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
                  "total_entries": 1,
                },
                "reference": "mega-menu",
                "title": "Main Menu",
                "updated_at": "2018-11-16T15:19:36Z",
              }
            }
          />
        </span>
        <div
          className="c-header__account"
        >
          <Image
            className="c-header__account-image"
            src={Object {}}
          />
          <Link
            href="/account/login"
          >
            Sign In
          </Link>
        </div>
        <div
          className="c-header__minibag c-minibag"
        >
          <span
            className="c-minibag__cart"
            onClick={[Function]}
          >
            <div
              className="c-minibag__cart-image"
            >
              <span
                className="c-minibag__cart-image-count"
              >
                0
              </span>
              <Image
                className="c-minibag__cart-image-icon"
                src={Object {}}
              />
            </div>
            <span
              className="c-minibag__cart-label"
            >
              Basket
            </span>
          </span>
        </div>
        <Minibag
          cart={
            Object {
              "line_items": Array [],
              "line_items_count": 0,
            }
          }
        />
        <span
          className="c-header__search"
        >
          <AlgoliaSearchBox(SearchBar) />
        </span>
      </div>
    </div>
    <span
      className="o-nav u-visible-d"
    >
      <NavBar
        menu={
          Object {
            "error": false,
            "id": "14",
            "loading": false,
            "menu_items": Array [
              Object {
                "canonical_path": "/trays",
                "id": "54",
                "meta_attributes": Object {},
                "published": true,
                "title": "Trays",
              },
              Object {
                "canonical_path": "/nonslip-trays",
                "id": "55",
                "meta_attributes": Object {},
                "published": true,
                "title": "Nonslip trays",
              },
              Object {
                "canonical_path": "/jugs",
                "id": "56",
                "meta_attributes": Object {},
                "published": true,
                "title": "Jugs",
              },
              Object {
                "canonical_path": "/filters",
                "id": "57",
                "meta_attributes": Object {},
                "published": true,
                "title": "Filters",
              },
              Object {
                "canonical_path": "/tea",
                "id": "58",
                "meta_attributes": Object {},
                "published": true,
                "title": "Tea",
              },
              Object {
                "canonical_path": "/saucers",
                "id": "60",
                "meta_attributes": Object {},
                "published": true,
                "title": "Saucers",
              },
              Object {
                "canonical_path": "/cups",
                "id": "59",
                "meta_attributes": Object {},
                "published": true,
                "title": "Cups",
              },
              Object {
                "canonical_path": "/coffee",
                "id": "61",
                "meta_attributes": Object {},
                "published": true,
                "title": "Coffee",
              },
              Object {
                "canonical_path": "/servers",
                "id": "63",
                "meta_attributes": Object {},
                "published": true,
                "title": "Servers",
              },
              Object {
                "canonical_path": "/mugs",
                "id": "62",
                "meta_attributes": Object {},
                "published": true,
                "title": "Mugs",
              },
            ],
            "pagination": Object {
              "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
              "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
              "page_count": 1,
              "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
              "total_entries": 1,
            },
            "reference": "mega-menu",
            "title": "Main Menu",
            "updated_at": "2018-11-16T15:19:36Z",
          }
        }
      />
    </span>
  </div>
  <div
    className="o-body"
  >
    Content
  </div>
  <div
    className="o-footer"
  >
    <Footer />
  </div>
</Fragment>
`;

exports[`renders the footer 1`] = `
<Fragment>
  <div
    className=""
  />
  <div
    className="o-footer"
  >
    <Footer />
  </div>
</Fragment>
`;

exports[`renders the header 1`] = `
<Fragment>
  <div
    className="o-header"
  >
    <div
      className="o-header__top"
    >
      <div
        className="o-header__top-wrapper"
      >
        <Logo
          className="o-header__logo"
        />
        <span
          className="o-nav u-hidden-d"
        >
          <NavBar
            menu={
              Object {
                "error": false,
                "id": "14",
                "loading": false,
                "menu_items": Array [
                  Object {
                    "canonical_path": "/trays",
                    "id": "54",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Trays",
                  },
                  Object {
                    "canonical_path": "/nonslip-trays",
                    "id": "55",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Nonslip trays",
                  },
                  Object {
                    "canonical_path": "/jugs",
                    "id": "56",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Jugs",
                  },
                  Object {
                    "canonical_path": "/filters",
                    "id": "57",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Filters",
                  },
                  Object {
                    "canonical_path": "/tea",
                    "id": "58",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Tea",
                  },
                  Object {
                    "canonical_path": "/saucers",
                    "id": "60",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Saucers",
                  },
                  Object {
                    "canonical_path": "/cups",
                    "id": "59",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Cups",
                  },
                  Object {
                    "canonical_path": "/coffee",
                    "id": "61",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Coffee",
                  },
                  Object {
                    "canonical_path": "/servers",
                    "id": "63",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Servers",
                  },
                  Object {
                    "canonical_path": "/mugs",
                    "id": "62",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Mugs",
                  },
                ],
                "pagination": Object {
                  "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
                  "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
                  "page_count": 1,
                  "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
                  "total_entries": 1,
                },
                "reference": "mega-menu",
                "title": "Main Menu",
                "updated_at": "2018-11-16T15:19:36Z",
              }
            }
          />
        </span>
        <div
          className="c-header__account"
        >
          <Image
            className="c-header__account-image"
            src={Object {}}
          />
          <Link
            href="/account/login"
          >
            Sign In
          </Link>
        </div>
        <div
          className="c-header__minibag c-minibag"
        >
          <span
            className="c-minibag__cart"
            onClick={[Function]}
          >
            <div
              className="c-minibag__cart-image"
            >
              <span
                className="c-minibag__cart-image-count"
              >
                0
              </span>
              <Image
                className="c-minibag__cart-image-icon"
                src={Object {}}
              />
            </div>
            <span
              className="c-minibag__cart-label"
            >
              Basket
            </span>
          </span>
        </div>
        <Minibag
          cart={
            Object {
              "line_items": Array [],
              "line_items_count": 0,
            }
          }
        />
        <span
          className="c-header__search"
        >
          <AlgoliaSearchBox(SearchBar) />
        </span>
      </div>
    </div>
    <span
      className="o-nav u-visible-d"
    >
      <NavBar
        menu={
          Object {
            "error": false,
            "id": "14",
            "loading": false,
            "menu_items": Array [
              Object {
                "canonical_path": "/trays",
                "id": "54",
                "meta_attributes": Object {},
                "published": true,
                "title": "Trays",
              },
              Object {
                "canonical_path": "/nonslip-trays",
                "id": "55",
                "meta_attributes": Object {},
                "published": true,
                "title": "Nonslip trays",
              },
              Object {
                "canonical_path": "/jugs",
                "id": "56",
                "meta_attributes": Object {},
                "published": true,
                "title": "Jugs",
              },
              Object {
                "canonical_path": "/filters",
                "id": "57",
                "meta_attributes": Object {},
                "published": true,
                "title": "Filters",
              },
              Object {
                "canonical_path": "/tea",
                "id": "58",
                "meta_attributes": Object {},
                "published": true,
                "title": "Tea",
              },
              Object {
                "canonical_path": "/saucers",
                "id": "60",
                "meta_attributes": Object {},
                "published": true,
                "title": "Saucers",
              },
              Object {
                "canonical_path": "/cups",
                "id": "59",
                "meta_attributes": Object {},
                "published": true,
                "title": "Cups",
              },
              Object {
                "canonical_path": "/coffee",
                "id": "61",
                "meta_attributes": Object {},
                "published": true,
                "title": "Coffee",
              },
              Object {
                "canonical_path": "/servers",
                "id": "63",
                "meta_attributes": Object {},
                "published": true,
                "title": "Servers",
              },
              Object {
                "canonical_path": "/mugs",
                "id": "62",
                "meta_attributes": Object {},
                "published": true,
                "title": "Mugs",
              },
            ],
            "pagination": Object {
              "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
              "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
              "page_count": 1,
              "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
              "total_entries": 1,
            },
            "reference": "mega-menu",
            "title": "Main Menu",
            "updated_at": "2018-11-16T15:19:36Z",
          }
        }
      />
    </span>
  </div>
  <div
    className="o-body"
  />
  <div
    className="o-footer"
  >
    <Footer />
  </div>
</Fragment>
`;

exports[`renders the navbar for non-checkout pages 1`] = `
<Fragment>
  <div
    className="o-header"
  >
    <div
      className="o-header__top"
    >
      <div
        className="o-header__top-wrapper"
      >
        <Logo
          className="o-header__logo"
        />
        <span
          className="o-nav u-hidden-d"
        >
          <NavBar
            menu={
              Object {
                "error": false,
                "id": "14",
                "loading": false,
                "menu_items": Array [
                  Object {
                    "canonical_path": "/trays",
                    "id": "54",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Trays",
                  },
                  Object {
                    "canonical_path": "/nonslip-trays",
                    "id": "55",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Nonslip trays",
                  },
                  Object {
                    "canonical_path": "/jugs",
                    "id": "56",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Jugs",
                  },
                  Object {
                    "canonical_path": "/filters",
                    "id": "57",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Filters",
                  },
                  Object {
                    "canonical_path": "/tea",
                    "id": "58",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Tea",
                  },
                  Object {
                    "canonical_path": "/saucers",
                    "id": "60",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Saucers",
                  },
                  Object {
                    "canonical_path": "/cups",
                    "id": "59",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Cups",
                  },
                  Object {
                    "canonical_path": "/coffee",
                    "id": "61",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Coffee",
                  },
                  Object {
                    "canonical_path": "/servers",
                    "id": "63",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Servers",
                  },
                  Object {
                    "canonical_path": "/mugs",
                    "id": "62",
                    "meta_attributes": Object {},
                    "published": true,
                    "title": "Mugs",
                  },
                ],
                "pagination": Object {
                  "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
                  "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
                  "page_count": 1,
                  "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
                  "total_entries": 1,
                },
                "reference": "mega-menu",
                "title": "Main Menu",
                "updated_at": "2018-11-16T15:19:36Z",
              }
            }
          />
        </span>
        <div
          className="c-header__account"
        >
          <Image
            className="c-header__account-image"
            src={Object {}}
          />
          <Link
            href="/account/login"
          >
            Sign In
          </Link>
        </div>
        <div
          className="c-header__minibag c-minibag"
        >
          <span
            className="c-minibag__cart"
            onClick={[Function]}
          >
            <div
              className="c-minibag__cart-image"
            >
              <span
                className="c-minibag__cart-image-count"
              >
                0
              </span>
              <Image
                className="c-minibag__cart-image-icon"
                src={Object {}}
              />
            </div>
            <span
              className="c-minibag__cart-label"
            >
              Basket
            </span>
          </span>
        </div>
        <Minibag
          cart={
            Object {
              "line_items": Array [],
              "line_items_count": 0,
            }
          }
        />
        <span
          className="c-header__search"
        >
          <AlgoliaSearchBox(SearchBar) />
        </span>
      </div>
    </div>
    <span
      className="o-nav u-visible-d"
    >
      <NavBar
        menu={
          Object {
            "error": false,
            "id": "14",
            "loading": false,
            "menu_items": Array [
              Object {
                "canonical_path": "/trays",
                "id": "54",
                "meta_attributes": Object {},
                "published": true,
                "title": "Trays",
              },
              Object {
                "canonical_path": "/nonslip-trays",
                "id": "55",
                "meta_attributes": Object {},
                "published": true,
                "title": "Nonslip trays",
              },
              Object {
                "canonical_path": "/jugs",
                "id": "56",
                "meta_attributes": Object {},
                "published": true,
                "title": "Jugs",
              },
              Object {
                "canonical_path": "/filters",
                "id": "57",
                "meta_attributes": Object {},
                "published": true,
                "title": "Filters",
              },
              Object {
                "canonical_path": "/tea",
                "id": "58",
                "meta_attributes": Object {},
                "published": true,
                "title": "Tea",
              },
              Object {
                "canonical_path": "/saucers",
                "id": "60",
                "meta_attributes": Object {},
                "published": true,
                "title": "Saucers",
              },
              Object {
                "canonical_path": "/cups",
                "id": "59",
                "meta_attributes": Object {},
                "published": true,
                "title": "Cups",
              },
              Object {
                "canonical_path": "/coffee",
                "id": "61",
                "meta_attributes": Object {},
                "published": true,
                "title": "Coffee",
              },
              Object {
                "canonical_path": "/servers",
                "id": "63",
                "meta_attributes": Object {},
                "published": true,
                "title": "Servers",
              },
              Object {
                "canonical_path": "/mugs",
                "id": "62",
                "meta_attributes": Object {},
                "published": true,
                "title": "Mugs",
              },
            ],
            "pagination": Object {
              "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
              "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
              "page_count": 1,
              "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
              "total_entries": 1,
            },
            "reference": "mega-menu",
            "title": "Main Menu",
            "updated_at": "2018-11-16T15:19:36Z",
          }
        }
      />
    </span>
  </div>
  <div
    className="o-body"
  />
  <div
    className="o-footer"
  >
    <Footer />
  </div>
</Fragment>
`;
